modules/globebrowsing/src/geojson/globegeometryfeature.cpp

```diff
@@ -0,0 +1,918 @@
+                                           GeoJsonProperties& defaultProperties,
+                                           GeoJsonOverrideProperties& overrideProperties)
+    : _globe(globe)
+    , _properties({ defaultProperties, overrideProperties })
+    , _lastHeightUpdateTime(std::chrono::system_clock::now())
+{}
+
+std::string GlobeGeometryFeature::key() const {
+    return _key;
+}
+
+void GlobeGeometryFeature::setOffsets(const glm::vec3& value) {
+    _offsets = value;
+}
+
+void GlobeGeometryFeature::initializeGL(ghoul::opengl::ProgramObject* pointsProgram,
+                                   ghoul::opengl::ProgramObject* linesAndPolygonsProgram)
+{
+    _pointsProgram = pointsProgram;
+    _linesAndPolygonsProgram = linesAndPolygonsProgram;
+
+    if (isPoints()) {
+        updateTexture(true);
+    }
+}
+
+void GlobeGeometryFeature::deinitializeGL() {
+    for (const RenderFeature& r : _renderFeatures) {
+        glDeleteVertexArrays(1, &r.vaoId);
+        glDeleteBuffers(1, &r.vboId);
+    }
+
+    _pointTexture = nullptr;
+}
+
+bool GlobeGeometryFeature::isReady() const {
+    bool shadersAreReady = (_linesAndPolygonsProgram != nullptr) &&
+        (_pointsProgram != nullptr);
+    bool textureIsReady = _hasTexture ? (_pointTexture != nullptr) : true;
+    return shadersAreReady && textureIsReady;
+}
+
+bool GlobeGeometryFeature::isPoints() const {
+    return _type == GeometryType::Point;
+}
+
+bool GlobeGeometryFeature::useHeightMap() const {
+    return _properties.altitudeMode() ==
+        GeoJsonProperties::AltitudeMode::RelativeToGround;
+}
+
+void GlobeGeometryFeature::updateTexture(bool isInitializeStep) {
+    std::string texture = "";
+    GlobeBrowsingModule* m = global::moduleEngine->module<GlobeBrowsingModule>();
+
+    if (!isInitializeStep && _properties.hasOverrideTexture()) {
+        // Here we don't necessarily have to update, since it should have been
+        // created at initialization. Do nothing
+        return;
+    }
+    else if (!_properties.pointTexture().empty()) {
+        texture = _properties.pointTexture();
+    }
+    else if (m->hasDefaultGeoPointTexture()) {
+        texture = m->defaultGeoPointTexture();
+    }
+    else {
+        // No texture => render without texture
+        _hasTexture = false;
+        _pointTexture = nullptr;
+        return;
+    }
+
+    if (isInitializeStep || !_pointTexture) {
+        _pointTexture = std::make_unique<TextureComponent>(2);
+        _pointTexture->setFilterMode(ghoul::opengl::Texture::FilterMode::AnisotropicMipMap);
+        _pointTexture->setWrapping(ghoul::opengl::Texture::WrappingMode::ClampToEdge);
+    }
+
+    std::filesystem::path texturePath = absPath(texture);
+    if (std::filesystem::is_regular_file(texturePath)) {
+        _hasTexture = true;
+        _pointTexture->loadFromFile(texture);
+        _pointTexture->uploadToGpu();
+    }
+    else {
+        LERROR(fmt::format("Trying to use texture file that does not exist"));
```

Comment:
Unsure. I actually don't think we should ever get to this point, but if we do it would not necessarily be a problem, as the `_pointTexture` would not be used if it does not exist

But Ill add the file name to the message either way